src/views/Login/Login.js

```diff
@@ -45,38 +44,40 @@ export default function SignIn() {
         event.preventDefault();
       };
 
-      const handleSubmit = (e) => {
+      const handleSubmit = async (e) => {
         e.preventDefault()
-        const savedUser = JSON.parse(localStorage.getItem('user'));
-        if (values.email === '') {
-          dispatch({type: SET_ERROR, payload: 'Email is required'})
-          toaster(auth.error, 'error')
-          return false
-        }
-        else if (values.email !== savedUser.Email) {
-          dispatch({type: SET_ERROR, payload: 'invalid Email or Password'})
-          toaster(auth.error, 'error')
-          return false
-        }
-        else if (values.password === '') {
-          dispatch({type: SET_ERROR, payload: 'Password is required'})
-          toaster(auth.error, 'error')
-          return false
-        }
-         else if (values.password !== savedUser.Password) {
-          dispatch({type: SET_ERROR, payload: 'invalid Email or Password'})
-          toaster(auth.error, 'error')
-          return false
-        }
-          dispatch({type: SET_AUTHENTICATION})
-          toaster('You are logged in', 'success')
-          setTimeout(() => {
-            history.push('/dashboard')
-          }, 4000)   
+        const body = {email:values.email,password:values.password};
+
+        const response =  await fetch('http://localhost:5000/user/signin',
+          {
+            method:'post',
+            headers:{"Content-Type":"Application/json"},
+            body:JSON.stringify(body)
+           });
+
+           const jsonData = await response.json();
+
+           if(jsonData.user !== undefined){
+
+               console.log(jsonData.user)
+               localStorage.setItem("userProfile",jsonData.user.id);
+               localStorage.setItem("userProfileToken",jsonData.token);
+               localStorage.setItem("userImageUrl",jsonData.user.user_image);
+                      
+               dispatch({type: SET_AUTHENTICATION})
+               toaster('You are logged in', 'success')
+                setTimeout(() => {
+                history.push('/dashboard');
```

Comment:
Good work for integrating the login but it's not fully done as it still depending on the other state which was there before. Try to login and after when you signed in try to refresh, it automatically redirect to the login page which is not how it might be.
since you have `token` the state should be depending on that token not `isAuthenticated` one which used dummy data.
whenever a token is not expired a user should be still logged in.

Reply:
The problem now is fixed